src/lti/outcome_request.py

```diff
@@ -51,15 +52,19 @@ def __init__(self, opts=defaultdict(lambda: None)):
                     "Invalid outcome request option: {}".format(key)
                 )
 
+        self.headers = CaseInsensitiveDict(headers)
```

Comment:
Since `headers` may be `None`, will this actually work? It's not typical for that to work for `dict`. I'd probably prefer this to be written:

```suggestion
        self.headers = CaseInsensitiveDict(headers or {})
```

Reply:
requests' CaseInsensitiveDict handles that case, but I've added the code you suggested for clarity.

Reply:
I've extended the test a bit further so there's a net improvement in the test coverage.